theora_image_transport/CMakeLists.txt

```diff
@@ -47,75 +46,74 @@ add_library(
   src/theora_subscriber.cpp
   src/manifest.cpp
 )
-
-include_directories(include
-  ${OpenCV_INCLUDE_DIRS}
-  ${PC_OGG_INCLUDE_DIRS}
-  ${PC_THEORA_INCLUDE_DIRS}
-  ${PC_THEORAENC_INCLUDE_DIRS}
-  ${PC_THEORADEC_INCLUDE_DIRS})
-
+target_compile_definitions(${LIBRARY_NAME} PRIVATE
+  ${PC_OGG_CFLAGS_OTHER}
+  ${PC_THEORA_CFLAGS_OTHER}
+  ${PC_THEORAENC_CFLAGS_OTHER}
+  ${PC_THEORADEC_CFLAGS_OTHER}
+)
+target_include_directories(${LIBRARY_NAME} PRIVATE
+  "$<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/include>"
+  "$<INSTALL_INTERFACE:include/${PROJECT_NAME}>"
+)
 target_link_libraries(${LIBRARY_NAME}
   ${OpenCV_LIBRARIES}
   ${PC_OGG_LIBRARIES}
   ${PC_THEORA_LIBRARIES}
   ${PC_THEORAENC_LIBRARIES}
   ${PC_THEORADEC_LIBRARIES}
+  "${cpp_typesupport_target}"
+  ${sensor_msgs_TARGETS}
+  cv_bridge::cv_bridge
+  image_transport::image_transport
+  pluginlib::pluginlib
+  rclcpp::rclcpp
 )
 
-ament_target_dependencies(${LIBRARY_NAME}
-  "image_transport"
-  "cv_bridge"
-  "rclcpp"
-  "pluginlib"
-  "sensor_msgs"
-)
-
-add_definitions(${PC_OGG_CFLAGS_OTHER}
-                ${PC_THEORA_CFLAGS_OTHER}
-                ${PC_THEORAENC_CFLAGS_OTHER}
-                ${PC_THEORADEC_CFLAGS_OTHER})
-
 add_executable(ogg_saver src/ogg_saver.cpp)
-target_link_libraries(ogg_saver ${PC_THEORA_LIBRARY} 
-                                ${PC_OGG_LIBRARY} 
-                                ${OpenCV_LIBRARIES} 
-                                ${PC_THEORAENC_LIBRARIES}
-                                ${PC_THEORADEC_LIBRARIES})
-
-ament_target_dependencies(ogg_saver
-  "rclcpp"
-  "rcutils"
+target_compile_definitions(ogg_saver PRIVATE
+  ${PC_OGG_CFLAGS_OTHER}
+  ${PC_THEORA_CFLAGS_OTHER}
+  ${PC_THEORAENC_CFLAGS_OTHER}
+  ${PC_THEORADEC_CFLAGS_OTHER}
 )
-
-rosidl_generate_interfaces(${PROJECT_NAME}
-  "msg/Packet.msg"
-  DEPENDENCIES
-    builtin_interfaces
-    std_msgs
-    sensor_msgs
-   ADD_LINTER_TESTS
+target_link_libraries(ogg_saver
+  ${PC_THEORA_LIBRARY}
+  ${PC_OGG_LIBRARY}
+  ${OpenCV_LIBRARIES}
```

Comment:
nit, I don't see OpenCV used in ogg_saver.cpp

Reply:
Fixed in ecf35d4.